src/meta/src/manager/catalog/fragment.rs

```diff
@@ -869,9 +869,14 @@ impl FragmentManager {
         let mut table_id_to_apply = HashSet::new();
         for (table_id, table_fragments) in map.iter() {
             for fragment in table_fragments.fragments.values() {
+                // CREATE SOURCE
                 if (fragment.get_fragment_type_mask() & FragmentTypeFlag::Source as u32) != 0 {
                     table_id_to_apply.insert(*table_id);
                 }
+                // CREATE TABLE WITH (connector='...')
+                if (fragment.get_fragment_type_mask() & FragmentTypeFlag::Mview as u32) != 0 {
+                    table_id_to_apply.insert(*table_id);
+                }
```

Comment:
Do you mean that we don't have a fragment with `Flag::Source` in a table job with `CREATE TABLE WITH (connector='...')`?

Reply:
The code here looks weird to me. https://github.com/risingwavelabs/risingwave/blob/468f7ec14f7e53d7c4d4af965eace50bf1d2fde7/src/meta/src/manager/catalog/fragment.rs#L877-L882

says that "source_id not found", but we don't actually check the source id at all in the code above. It appears to just filter out all streaming jobs without a `Source` fragment (which means jobs on MVs).

Reply:
cc @tabVersion can you PTAL at @BugenZhao 's remarks?